crev-lib/src/local.rs

```diff
@@ -243,37 +382,108 @@ impl Local {
         id.save_to(&path)
     }
 
-    pub fn git_init_proof_dir(&self, git_https_url: &str) -> Result<()> {
-        let git_url = https_to_git_url(git_https_url);
+    fn init_remote_github_proof_repo(
+        &self,
+        proof_dir: &Path,
+        github_username: &str,
+        git_https_url: &str,
+        git_ssh_url: &Option<String>,
+    ) -> Result<()> {
+        fs::create_dir_all(&proof_dir)?;
+
+        let repo = git2::Repository::init(&proof_dir)?;
+        eprintln!("Initialized empty local git repository: {}", proof_dir.display());
+
+        if let Some(git_ssh_url) = git_ssh_url {
+            repo.remote_set_url("origin", &git_ssh_url)?;
+        } else {
+            repo.remote_set_url("origin", &git_https_url)?;
+        }
+
+        eprint!("Enter Github password: ");
+        let github_password = rpassword::read_password()?;
+
+        let repo_name = parse_git_repository_name(git_https_url).unwrap_or("crev-proofs".to_string());
+        let response = setup_remote_github_repository(
+            &github_username,
+            &github_password,
+            &repo_name);
+        check_github_response(&response);
 
-        let proof_dir =
-            self.get_proofs_dir_path_for_url(&Url::new_git(git_https_url.to_owned()))?;
+        let readme_file_name = Path::new("README.md");
```

Comment:
I've just added a function creating `README_USING_THIS_REPO.md`. Use it instead. `README.md` is supposed to  be personalized, for the user.